backend/src/main/resources/db/migration/V1.2__add_table_update_tracker.sql

```diff
@@ -0,0 +1,42 @@
+-- Create the table_update_tracker table
+CREATE TABLE table_update_tracker (
+    table_name TEXT PRIMARY KEY,
+    last_updated TIMESTAMP DEFAULT timezone('UTC', CURRENT_TIMESTAMP)
+);
+
+-- Function to update the last_updated timestamp
+CREATE OR REPLACE FUNCTION update_table_tracker()
+RETURNS TRIGGER AS $$
+BEGIN
+    IF TG_TABLE_NAME != 'table_update_tracker' THEN
+        INSERT INTO table_update_tracker (table_name, last_updated)
+        VALUES (TG_TABLE_NAME, timezone('UTC', CURRENT_TIMESTAMP))
+        ON CONFLICT (table_name)
+        DO UPDATE SET last_updated = timezone('UTC', CURRENT_TIMESTAMP);
+    END IF;
+    RETURN NULL;
+END;
+$$ LANGUAGE plpgsql;
+
+-- Function to create a trigger for a single table
+CREATE OR REPLACE FUNCTION create_update_trigger_for_table(table_name TEXT)
+RETURNS VOID AS $$
+BEGIN
+    IF table_name != 'table_update_tracker' THEN
+        EXECUTE format('
+            CREATE OR REPLACE TRIGGER update_tracker_trigger
+            AFTER INSERT OR UPDATE OR DELETE OR TRUNCATE ON %I
+            FOR EACH STATEMENT
+            EXECUTE FUNCTION update_table_tracker()', table_name);
+    END IF;
+END;
+$$ LANGUAGE plpgsql;
+
+SELECT create_update_trigger_for_table('sequence_entries');
```

Comment:
Most tables (when reading) use the `sequence_entries_view` (which depends on which depends on `sequence_entries`, `sequence_entries_preprocessed_data`, `current_processing_pipeline` and `all_external_metadata`), so would it be useful to directly track changes to the view instead? Otherwise, the backend would need to calculate it itself which doesn't seem great because so far, it just used the view without the need to know on which tables it is based on.

Reply:
As views aren't written I don't think that works - or how would you attach a trigger to a view?

>  Otherwise, the backend would need to calculate it itself which doesn't seem great because so far, 

The calculation is essentially (pseudocode)
```
max(table_update_tracker.last_update)
```

where one excludes maybe `sequence_entries_preprocessed_data` (or fixes that it doesn't update all the time).

Do you mean you'd construct a new view `last_update_to_released_data` that simply contains the result of the calculation to put the code into sql as opposed to in the backend? I guess one could do that.

Reply:
Either a view as you suggested, or let any change to one of the four tables trigger an update of the entry of the view.

Reply:
The goal is simply to keep the view-related logic together in one place and not divide it up between the SQL scripts and the backend.